drivers/ma_yya_report/app/models/ma_yya_report/universe_calculator.rb

```diff
@@ -0,0 +1,199 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module MaYyaReport
+  class UniverseCalculator
+    include Filter::FilterScopes
+
+    def initialize(filter)
+      @filter = filter
+    end
+
+    def client_ids
+      # We don't use tap here to allow for '+='
+      client_ids = Set.new
+      client_scope.find_in_batches do |batch|
+        clients = for_batch(batch)
+        client_ids += clients.keys
+      end
+      client_ids
```

Comment:
This seems like a really expensive way to get client_ids.  Maybe I'm missing something, but you fetch all of the clients in the scope, then fetch all of their enrollments, then fill in all the data of a massive hash keyed on client_id, and throw away the values.
I _think_
```suggestion
      client_scope.distinct.pluck(:id)
```
would accomplish the same thing.

Reply:
Good catch! I think the expectation was the `for_batch` might not return all the clients in the scope based on the data within the scope, but that doesn't appear to be the case.